modules/core/src/main/java/org/apache/ignite/internal/cache/query/index/sorted/inline/types/NullableInlineIndexKeyType.java

```diff
@@ -167,9 +174,9 @@ protected NullableInlineIndexKeyType(IndexKeyType type, short keySize) {
      * @param pageAddr Page address.
      * @param off Offset.
      *
-     * @return Inline value or {@code null} if value can't be restored.
+     * @return Inline value.
      */
-    protected abstract @Nullable T get0(long pageAddr, int off);
+    protected abstract T get0(long pageAddr, int off);
```

Comment:
There is still a code, checking the null: `IndexQueryCriteriaClosure#rowIsOutOfRange()`. Should it be fixed? There is no null returns any more?

Reply:
`rowIsOutOfRange` checks `get` method. `get` can be null, but `get0` can't.